### DISCUSSION TITLE
Cart Transform: Line Item Properties in Input

### DISCUSSION THREAD

**Discussion by danielfv** [Shopify/function-examples#327] [Feedback: Cart Transform API]
Hi there,

It would be awesome to have access to the line item properties in the cart on this function. 

Thanks!

**Comment by ignacio-chiazzo** [comment]
Hi! Could you elaborate more on that?

 [`Cart` and `CartLine`](https://github.com/Shopify/function-examples/blob/main/checkout/javascript/cart-transform/default/schema.graphql#L90) allow querying attributes by via the API.

**Reply by danielfv** [reply]
Hi!

What I mean is to receive as input the line item properties ([https://shopify.dev/docs/api/liquid/objects/line_item#line_item-properties](https://shopify.dev/docs/api/liquid/objects/line_item#line_item-properties)

So based on this we can take actions with cart transform.

**Reply by nickwesselman** [reply]
Hi @danielfv -- As mentioned [in the docs](https://shopify.dev/docs/api/functions/reference/cart-transform/graphql/common-objects/cartline), cart line attributes are the same as line item properties in Liquid.
